perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/AuthzResolverMethod.java

```diff
@@ -250,6 +270,15 @@ public AuthzRoles call(ApiCaller ac, Deserializer parms) throws PerunException {
 	 * @exampleParam role "VOADMIN"
 	 * @exampleParam complementaryObject { "id" : 123 , "name" : "My testing VO" , "shortName" : "test_vo" , "beanName" : "Vo" }
 	 */
+	/*#
+	 * Set role for users.
+	 *
+	 * IMPORTANT: Refresh authz only if user in session is affected.
+	 *
+	 * @param role String Role which will be set for given users ( FACILITYADMIN | GROUPADMIN | PERUNADMIN | RESOURCEADMIN | RESOURCESELFSERVICE | SPONSOR | TOPGROUPCREATOR | VOADMIN | VOOBSERVER | PERUNOBSERVER | SECURITYADMIN | CABINETADMIN )
+	 * @param users int[] <code>ids</code> of users for which is the role set
+	 * @exampleParam role "VOADMIN"
+	 */
```

Comment:
So, generally there is no longer any check, that role requires some complementary object and we can expect more "group managers without any group" or what? If there is no check, then it might fail in ``AuthzResolverBlImpl.unsetRole(sess, user, complementaryObject, role)`` for role ``SPONSOR`` where code fetches beanName from complementaryObject. There might be more places like this.

Reply:
Rules for that are defined in the perun-roles.yml. So for example, it is possible to set/unset role without a complementary object for PERUNADMIN, PERUNOBSERVER, CABINETADMIN. However, it is not possible to set/unset role without a complementary object for GROUPADMIN, VOADMIN, SPONSOR etc.
This is checked in the business logic of set/unset role methods.

Reply:
Ok, that makes sense. It just might be confusing in javadoc, since we do not list correct combinations, but only all possible. But since its technically configurable per-instance we probably can't do anything about it.